src/SortableTrait.php

```diff
@@ -219,10 +221,41 @@ public function moveToEnd()
         $this->$orderColumnName = $maxOrder;
         $this->save();
 
-        static::where($this->getKeyName(), '!=', $this->id)
+        static::applySortableGroup($this->newQuery(), $this)->where($this->getKeyName(), '!=', $this->id)
             ->where($orderColumnName, '>', $oldOrder)
             ->decrement($orderColumnName);
 
         return $this;
     }
+
+    /**
+     * @param QueryBuilder        $query
+     * @param Model|SortableTrait $model
+     *
+     * @return QueryBuilder
+     */
+    protected static function applySortableGroup($query, $model)
+    {
+        $sortableGroupField = static::getSortableGroupField();
+
+        if (is_array($sortableGroupField)) {
```

Comment:
Personally, I would do this: if the `$sortableGroupField` is string, convert it into an array, therefore you can remove `elseif` check.